adafruit_itertools/__init__.py

```diff
@@ -342,17 +342,34 @@ def islice(p, start, stop=(), step=1):
     if stop == ():
         stop = start
         start = 0
+    if stop is not None and stop < 0:
+        raise ValueError("Stop for islice must be None or a non-negative integer")
+    if start < 0:
+        raise ValueError("Start for islice must be a non-negative integer")
+    if step <= 0:
+        raise ValueError("Step for islice must be a positive integer")
```

Comment:
Let's make these messages somewhat shorter to save RAM. Don't need to mention it's islice: that will be clear from the context. And this isn't even checking if they args are integers. This style is similar to what we use inside CIrcuitPython.
```suggestion
    if stop is not None and stop < 0:
        raise ValueError("stop must be None or >= 0")
    if start < 0:
        raise ValueError("start must be >= 0")
    if step <= 0:
        raise ValueError("step must be > 0")
```